### PR TITLE
feat: add support for JSON format in output

### PR DESCRIPTION
**Pull request by nryn**
centos-package-cron does not currently have the ability to output results in a machine-readable format (e.g. for automated checks).

This pull request implements a `-j / --json` flag to format output as JSON.

## Current output
```
The following security advisories exist for installed packages:

Advisory ID: adv id
Severity: Important
Packages:
* libgcrypt-1.5.3-4.el7
References:



The following packages are available for updating:

libgcrypt-1.5.3-4.el7 from updates



Change logs for available package updates:

libgcrypt-1.5.3-4.el7
stuff
```

## JSON output
```
{
    'advisories':[{
        "advisory_id": "adv id",
        "packages": ["libgcrypt-1.5.3-4.el7"],
        "severity": "Important"
    }],
    'updates':[{
        "name":"libgcrypt-1.5.3-4.el7",
        "source":"updates",
        "changelog": "stuff"
    }]
}

```

## Contributors

Alex Wilson (@mrwilson)
Narayan O'Hanlon (@nryn)
Sarah Young (@sarahseewhy)
Stephen Geller (@stephengeller)

### CODE REVIEW

**Comment by mrwilson** [comment]
The tests are passing against the [centos7](https://travis-ci.org/wied03/centos-package-cron/jobs/330883468#L1259), and passed locally for us, but the builds in Travis appear to have either python or rpmbuild errors.

**Comment by wied03** [comment]
Thanks! Let me get the baseline tests working and then we can merge this.

**Comment by wied03** [comment]
@mrwilson - Got tests working (simplified things a lot, left out CentOS6 from automated tests for now). Can you rebase this onto master and then we can go from there?

**Comment by mrwilson** [comment]
@wied03 - great! Rebased and green :)

File: .gitignore [hunk 1]
```diff
@@ -13,3 +13,4 @@ centos_package_cron_src.tgz
 centos-package-cron.spec
 .bundle
 Gemfile.lock
+deploy.sh
```
Comment (wied03) [inline .gitignore]: @mrwilson or @nryn - Is this an accidental commit?

Comment (mrwilson) [reply .gitignore]: This is accidental, we'll roll this back.

**Comment by FTBZ** [comment]
Thanks a lot for these changes! This opens up a lot of possibility.

**Pull request merged**